src/transformers/pipelines/document_question_answering.py

```diff
@@ -0,0 +1,460 @@
+                # return_overflowing_tokens=True,
+                **tokenizer_kwargs,
+            )
+
+            if "pixel_values" in image_features:
+                encoding["image"] = image_features.pop("pixel_values")
+
+            # TODO: For now, this should always be num_spans == 1 given the flags we've passed in above, but the
+            # code is written to naturally handle multiple spans at the right time.
+            num_spans = len(encoding["input_ids"])
+
+            # p_mask: mask with 1 for token than cannot be in the answer (0 for token which can be in an answer)
+            # We put 0 on the tokens from the context and 1 everywhere else (question and special tokens)
+            # This logic mirrors the logic in the question_answering pipeline
+            p_mask = [[tok != 1 for tok in encoding.sequence_ids(span_id)] for span_id in range(num_spans)]
+            for span_idx in range(num_spans):
+                input_ids_span_idx = encoding["input_ids"][span_idx]
+                # keep the cls_token unmasked (some models use it to indicate unanswerable questions)
+                if self.tokenizer.cls_token_id is not None:
+                    cls_indices = np.nonzero(np.array(input_ids_span_idx) == self.tokenizer.cls_token_id)[0]
+                    for cls_index in cls_indices:
+                        p_mask[span_idx][cls_index] = 0
+
+            # For each span, place a bounding box [0,0,0,0] for question and CLS tokens, [1000,1000,1000,1000]
+            # for SEP tokens, and the word's bounding box for words in the original document.
+            if "boxes" not in tokenizer_kwargs:
+                bbox = []
+                for batch_index in range(num_spans):
+                    for i, s, w in zip(
+                        encoding.input_ids[batch_index],
+                        encoding.sequence_ids(batch_index),
+                        encoding.word_ids(batch_index),
+                    ):
+                        if s == 1:
+                            bbox.append(boxes[w])
+                        elif i == self.tokenizer.sep_token_id:
+                            bbox.append([1000] * 4)
+                        else:
+                            bbox.append([0] * 4)
+
+                if self.framework == "tf":
+                    raise ValueError("Unsupported: Tensorflow preprocessing for DocumentQuestionAnsweringPipeline")
+                elif self.framework == "pt":
+                    encoding["bbox"] = torch.tensor([bbox])
+
+            word_ids = [encoding.word_ids(i) for i in range(num_spans)]
+
+        return {
+            **encoding,
+            "p_mask": p_mask,
+            "word_ids": word_ids,
+            "words": words,
+        }
+
+    def _forward(self, model_inputs):
+        p_mask = model_inputs.pop("p_mask", None)
+        word_ids = model_inputs.pop("word_ids", None)
+        words = model_inputs.pop("words", None)
+
+        if self.model_type == ModelType.Donut:
```

Comment:
I wouldn't name it Donut. This is not because it's donut, it's because it's an Encoder/Decoder model I think (We can just change the name of this enum.)

Pipeline *are* model agnostic, they really should rely of `AutoModeForXXX`(which can be implemented by many more models than donut for instance).